### PR TITLE
DotNet SDK Default Paths

### PR DESCRIPTION
**Pull request by baronfel**
First pass at unifying the default paths to the system-wide install locations.

Outstanding questions:
* Do we want to take on global.json enhancements here?
* If so, where should global.json be consulted?

Related to #1961

### CODE REVIEW

**Comment by baronfel** [comment]
This failed during the packaging steps for the exact reasons mentioned in the attached discussion: The packaging was done with the `C:\Program Files\dotnet` version of dotnet and not the local one.

**Comment by baronfel** [comment]
Ah I see, the install needs to pass that install dir arg, yeah

**Comment by matthid** [comment]
No actually I think the installer should install in users directory as otherwise this only leads to permission problems. So we basically have two "defaults", one where we look for the global installation and one where we look for user-local installations.

**Comment by baronfel** [comment]
If we do that then we have to then do additional probing across the two locations to check which one (or both?) has the intended version. Which again, is hard to reconcile with PATH. Are we sure that permissions will be a big deal?

**Comment by matthid** [comment]
Maybe there is another bug as it seems to ignore the install parameter at the moment

**Comment by matthid** [comment]
> Are we sure that permissions will be a big deal?

Yes there will be people on corporate environments who will thank you

**Comment by matthid** [comment]
ping :)

File: src/legacy/Test.FAKECore/PackageMgt/FeedSpecs.cs [hunk 1]
```diff
@@ -37,7 +37,10 @@ public class when_discovering_the_lastest_FAKE_package
             () => _package.Published.Year.ShouldBeGreaterThanOrEqualTo(2012);
 
         It should_contain_the_license_url = 
-            () => _package.LicenseUrl.ShouldStartWith("https://github.com/fsharp/Fake/blob/master/License.txt") && _package.LicenseUrl.ShouldEndWith("License.txt");
```
Comment (matthid) [inline src/legacy/Test.FAKECore/PackageMgt/FeedSpecs.cs]: Sorry, please just merge release/next I already fixed that (and actually made the same mistake :D )

Comment (baronfel) [reply src/legacy/Test.FAKECore/PackageMgt/FeedSpecs.cs]: Hahaha we would make the worst C# devs 😄

**Comment by matthid** [comment]
@baronfel @TheAngryByrd @kblohm What do you think about https://github.com/fsharp/FAKE/pull/1963/commits/30bf7e8e28d00377576ff21a4b16bc8a73eabbe2 ? The basic idea is that the "global.json" topic is orthogonal to installing the SDK with FAKE.

This commit 
- introduces a `install FromGlobalJson`-Api which allows to manage the version in global.json
- writes `global.json` before calling dotnet SDK (and deletes it afterwards) to make the SDK choose the correct version
- Errors out if global.json exists and doesn't match
- We search and try more directories for installed SDK versions

**Comment by kblohm** [comment]
That does probably solve a lot of the versioning problems 👍 
Some comments
* You might want to add `withVersion` functions similair to `withEnvironment` to all the options or at least to the options-module (`preferSystemInstall` is also in the options-module and i feel like these kinda belong in the same category)
* It looks like this would not work when there is a global.json file already that is used for something else then specifying the dotnet version (for example https://github.com/onovotny/MSBuildSdkExtras#using-the-sdk)
* It might be nice to have the option to create a `global.json` when installing a version with fake, so that it gets used for everything including VS etc.

**Comment by matthid** [comment]
> You might want to add `withVersion`

I'm not sure you want to use the `Version` option directly, my suggestion would be to always use the result of the install API.

>  when there is a global.json file 

We don't replace an existing file so we either exit with an error or run with the wrong version ... Good catch

> It might be nice to have the option to create a global.json when installing a version with fake

For now it's easier to let users create it, because for this we would need a way to figure out if we wrote the file (such that we can overwrite it later if needed). But yeah makes sense.

**Pull request merged**